docs/usage.md

```diff
@@ -175,30 +175,58 @@ contrasts:
+The necessary fields in order are:
+
+- `id` - an arbitrary identifier, will be used to name contrast-wise output files
+- `comparison`(respectively):
+- - `variable` - which column from the observations information will be used to define groups
+- - `reference` - the base/ reference level for the comparison. If features have higher values in this group than target they will generate negative fold changes
+- - `target` - the target/ non-reference level for the comparison. If features have higher values in this group than the reference they will generate positive fold changes
+- `blocking_factors` - Any additional variables (also observation columns) that should be modelled alongside the contrast variable
+- `exclude_samples_col` and `exclude_samples_values` - the former being a valid column in the samples sheet, the latter a list of values in that column which should be used to select samples prior to differential modelling. This is helpful where certain samples need to be excluded prior to analysis of a given contrast.
+
+Alternatively, the YAML contrasts also supports formula based model definitions for tools such as `VARIANCEPARTITION_DREAM`:
 
 ```yaml
 contrasts:
   - id: condition_control_treated
     formula: "~ condition"
-    comparison: ["condition", "control", "treated"]
+    make_contrasts_str: "conditiontreated"
   - id: condition_control_treated_blockrep
     formula: "~ condition + replicate"
-    comparison: ["condition", "control", "treated"]
+    make_contrasts_str: "conditiontreated"
 ```
 
-The necessary fields in order are:
+The necessary fields in order are::
 
-- `id` - an arbitrary identifier, will be used to name contrast-wise output files
-- `comparison`(respectively):
-- - `variable` - which column from the observations information will be used to define groups
-- - `reference` - the base/ reference level for the comparison. If features have higher values in this group than target they will generate negative fold changes
-- - `target` - the target/ non-reference level for the comparison. If features have higher values in this group than the reference they will generate positive fold changes
+- `formula` - A string representation of the model formula. It is used to build the design matrix.
+- `make_contrasts_str` - An explicit literal contrast string (e.g., "treatmenthND6 - treatmentmCherry") that is passed directly to [`limma::makeContrasts()`](https://rdrr.io/bioc/limma/man/makeContrasts.html) in `VARIANCEPARTITION_DREAM`. The parameter names must be syntactically valid variable names in R (see [`make.names`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/make.names.html)). This field provides full control for complex designs. Requires `formula`.
 
-You can optionally supply:
+> [!WARNING]
+> Formula-based contrasts are currently only supported by `VARIANCEPARTITION_DREAM`. They **do not work** with tools like `DESEQ2` or base `LIMMA` implementations yet.
```

Comment:
I guess this is soon to change?

Reply:
Yes, but for the moment we'll keep this way because I am not sure if the new implementations will make it to next release.